WooCommerce/Classes/POS/Card Present Payments/CardPresentPaymentEvent.swift

```diff
@@ -0,0 +1,8 @@
+import Foundation
+
+enum CardPresentPaymentEvent {
+    case idle
+    case showAlert(_ alertViewModel: CardPresentPaymentsModalViewModel)
```

Comment:
So far, I've avoided replacing `CardPresentPaymentsModalViewModel` when we emit a `showAlert` event... but I'm in two minds. The problem here is that we'll only get that generic view model from the underlying code, and things like `topTitle` and `bottomSubtitle` will be difficult to give much semantic meaning to. 

We could duplicate it quite closely, and translate to that, I suppose... at least we could avoid sharing the UIViewController tap handlers and UIImage.

Reply:
Is this related to the original splitting of `CardPresentPaymentsModalViewModel` protocol to modal **content** and **actions** in the [experiment PR](https://github.com/woocommerce/woocommerce-ios/pull/12760/files#diff-012837b4be8f858490b971f757f3c4a33a63e877fc0e161150973112c30b15c2L5)? Or did you mean to replace `CardPresentPaymentsModalViewModel` with something else?

Reply:
Yes, it's related to that. Perhaps we could split it to three things:
- content
- UIKit actions
- generic actions (which we'll use)

Then we could have a typealias for the first and third of these, which is what we'll require here, but still fulfil that with the `CardPresentPaymentsModalViewModel`.

The other option is to transform the `CardPresentPaymentsModalViewModel` to a new model which we add here as well, but that has the problem of the semantic meaning of the various titles being unclear.